server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -145,7 +142,18 @@ private SourceFieldMapper(boolean enabled, String[] includes, String[] excludes)
         this.includes = includes;
         this.excludes = excludes;
         final boolean filtered = CollectionUtils.isEmpty(includes) == false || CollectionUtils.isEmpty(excludes) == false;
-        this.filter = enabled && filtered ? XContentMapValues.filter(includes, excludes) : null;
+        if (enabled && filtered) {
+            final XContentParserConfiguration parserConfig = XContentParserConfiguration.EMPTY.withFiltering(Set.of(includes), Set.of(excludes));
+            this.filter = (sourceBytes, contentType) -> {
+                BytesStreamOutput streamOutput = new BytesStreamOutput(Math.min(1024, sourceBytes.length()));
+                XContentBuilder builder = new XContentBuilder(XContentType.JSON.xContent(), streamOutput);
+                XContentParser parser = XContentType.JSON.xContent().createParser(parserConfig, sourceBytes.streamInput());
+                builder.copyCurrentStructure(parser);
+                return BytesReference.bytes(builder);
```

Comment:
I believe we need to use the old map-based filtering mechanism if there are wildcards in the excludes. #81634 surfaces at least one case. I believe there are others. It's also similar to #63842.

Reply:
I'd been holding off doing this kind of work because I've been busy. And because I've wanted to upgrade Jackson and resolve #63842.

Reply:
Thanks, I add a check before creating filter. Map-based filtering will be used if there are some wildcard in `exclude`. And I will keep track of bad cases together with @weizijun .